core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/FulltextSearchQueryUtils.scala

```diff
@@ -0,0 +1,151 @@
+import java.sql.Timestamp
+import java.text.{ParseException, SimpleDateFormat}
+import java.util.concurrent.TimeUnit
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+object FulltextSearchQueryUtils {
+
+  def getOrderFields(
+      specificResourceType: String,
+      searchQueryParams: SearchQueryParams
+  ): List[OrderField[_]] = {
+    // Regex pattern to extract column name and order direction
+    val pattern = "(Name|CreateTime|EditTime)(Asc|Desc)".r
+
+    searchQueryParams.orderBy match {
+      case pattern(column, order) =>
+        val field = getColumnField(specificResourceType, column)
+        field match {
+          case Some(value) =>
+            List(order match {
+              case "Asc"  => value.asc()
+              case "Desc" => value.desc()
+            })
+          case None => List()
+        }
+      case _ => List() // Default case if the orderBy string doesn't match the pattern
+    }
+  }
+
+  // Helper method to map column names to actual database fields based on resource type
+  private def getColumnField(resourceType: String, columnName: String): Option[Field[_]] = {
+    Option((resourceType, columnName) match {
+      case ("workflow", "Name")       => WORKFLOW.NAME
+      case ("workflow", "CreateTime") => WORKFLOW.CREATION_TIME
+      case ("workflow", "EditTime")   => WORKFLOW.LAST_MODIFIED_TIME
+      case ("project", "Name")        => PROJECT.NAME
+      case ("project", "CreateTime")  => PROJECT.CREATION_TIME
+      // Assuming PROJECT does not have a separate LAST_MODIFIED_TIME and uses CREATION_TIME for edits
+      case ("project", "EditTime") => PROJECT.CREATION_TIME
+      case ("file", "Name")        => FILE.NAME
+      case ("file", "CreateTime")  => FILE.UPLOAD_TIME
+      // Assuming FILE uses UPLOAD_TIME for edit time as well
+      case ("file", "EditTime") => FILE.UPLOAD_TIME
+      case _                    => null // Default case for unmatched resource types or column names
+    })
+  }
+
+  def getFulltextSearchConditions(
+      keywords: Seq[String],
+      fields: List[Field[String]]
+  ): Condition = {
+    if (keywords.nonEmpty && fields.isEmpty) {
+      return noCondition()
+    }
+    var result = noCondition()
+    val fullFieldNames = fields.map(_.toString.replace("\"", ""))
+    val indexedCompoundFields = fullFieldNames.mkString(",")
+    for (key: String <- keywords) {
+      if (key != "") {
+        val words = key.split("\\s+")
+        var conditionForKeyword = condition(
+          s"MATCH($indexedCompoundFields) AGAINST('${words.mkString("+", " +", "")}' IN BOOLEAN MODE)",
+          key
+        )
+        // perform exact "contains" to improve the quality of search results
+        // if the following has a huge performance overhead, consider removing the LIKE part.
+        for (fieldName <- fullFieldNames) {
+          conditionForKeyword = conditionForKeyword.or(s"$fieldName LIKE '%$key%'")
+        }
+        result = result.and(conditionForKeyword)
```

Comment:
I understand that you want to add "LIKE" condition. two comments:
1. this "LIKE" is not part of a full-text search. it is a substring search instead. I suggest we extract this out to a dedicated function, and the user can decide how to use multiple search methods (i.e., full-text only, substring only, full-text OR substring, full-text AND substring).
2. different databases may support “LIKE” in different ways. MySQL and SQLite by default "LIKE" will be case insensitive; PostgreSQL will treat "LIKE" as case sensitive and "ILIKE" as case insensitive. Maybe consider making this a substring match an abstract method and let different database instances to override the actual SQL pattern?

Reply:
I suggest we introduce multiple implementations in another PR for the following reasons:
1. It is not clear to me how to abstract this function. 
2. Now we only have mysql as our DB. We will not use other implementations even if I write them and we don't know how to test them.